### PR TITLE
Settings: Root

### PR DESCRIPTION
**Pull request by mhayes853**
Implements the main settings screen (minus navigation). Overall, the screen is straightforward.
IMG_0101

For the account info link, I made sure that it doesn't appear when the user is not signed in. To do this, I updated the hooks inside `UserSession.tsx` and used the `IfAuthenticated` component to decide whether or not to render the link. The `UserSessionProvider` now expects a function to return the current `UserSession` rather than a boolean that says whether or not the user is signed in.

We'll probably add a lot more to the screen in the near future, and we could possibly shrink the amount of padding between sections (this would go for all settings screens).

https://trello.com/c/oyp2MSvn

### CODE REVIEW

**Comment by tifapp** [comment]
Linked Ticket: https://trello.com/c/oyp2MSvn

**Comment by seanim0920** [comment]
Looks clean and inviting. Nice design.

File: settings-boundary/RootSettings.tsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+  areNotificationsEnabled
+}: MainScreenLinksSectionProps) => (
+  <SettingsCardSectionView>
+    <SettingsNavigationLinkView
+      iconName="settings"
+      iconBackgroundColor={AppStyles.black}
+      title="General"
+      onTapped={() => onSettingsScreenLinkTapped("generalSettings")}
+    />
+    <IfAuthenticated
```
Comment (seanim0920) [inline settings-boundary/RootSettings.tsx:54]: Nice that we can easily toggle visibility of the screens like this

File: settings-boundary/RootSettings.tsx [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+      title="Events"
+      onTapped={() => onSettingsScreenLinkTapped("eventSettings")}
+    />
+    <SettingsNavigationLinkView
+      iconName="notifications"
+      iconBackgroundColor={AppStyles.yellow}
+      title="Notifications"
+      rightAccessory={
+        <BodyText style={styles.notificationsText}>
+          {areNotificationsEnabled ? "On" : "Off"}
```
Comment (seanim0920) [inline settings-boundary/RootSettings.tsx:82]: Can we call it "areSystemNotificationsEnabled", I was unclear at first which notification this was referring to

File: event-details-boundary/Menu.test.tsx [hunk 3]
```diff
@@ -188,6 +197,10 @@ describe("EventDetailsMenu tests", () => {
       })
     })
 
+    const ensureUserIsSignedIn = () => {
```
Comment (seanim0920) [inline event-details-boundary/Menu.test.tsx:200]: nit - can we call it something like "useSignedInUser", "ensure" sounds like we're checking a condition

**Review by seanim0920 — APPROVED** [review]

**Pull request merged**